pkg/liqo-controller-manager/namespaceMap-controller/manage_namespacemap_controller_finalizer.go

```diff
@@ -0,0 +1,52 @@
+package namespacemapctrl
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/klog"
+	ctrlutils "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+)
+
+const (
+	namespaceMapControllerFinalizer = "namespacemap-controller.liqo.io/finalizer"
+)
+
+// SetNamespaceMapControllerFinalizer adds NamespaceMapControllerFinalizer to
+// a NamespaceMap, if it is not already there.
+func (r *NamespaceMapReconciler) SetNamespaceMapControllerFinalizer(ctx context.Context,
+	nm *mapsv1alpha1.NamespaceMap) error {
+	if !ctrlutils.ContainsFinalizer(nm, namespaceMapControllerFinalizer) {
+		patch := nm.DeepCopy()
+		ctrlutils.AddFinalizer(nm, namespaceMapControllerFinalizer)
+		if err := r.Patch(ctx, nm, client.MergeFrom(patch)); err != nil {
+			klog.Errorf("%s --> Unable to add finalizer to the NamespaceMap '%s'", err, nm.GetName())
+			return err
+		}
+		klog.Infof("Finalizer correctly added on NamespaceMap '%s'", nm.GetName())
+	}
+	return nil
+}
+
+// TryToRemoveNamespaceMapControllerFinalizer remove the NamespaceMapController finalizer if it is possible.
+// If the finalizer is correctly removed returns "true".
+func (r *NamespaceMapReconciler) TryToRemoveNamespaceMapControllerFinalizer(ctx context.Context,
+	nm *mapsv1alpha1.NamespaceMap) (bool, error) {
+	removed := false
+	if len(nm.Status.CurrentMapping) == 0 {
```

Comment:
I would personally move this check to the caller, as I feel the function name is still difficult to understand. The workflow, in that case, would be clearer.

Reply:
you are right, thank you